docs/operator.rst

```diff
@@ -673,6 +673,52 @@ Then you will be able to specify the tag ``foo`` or ``bar`` as an argument of th
 either the content of the first or the second channel. Multiple tag names can be specified separating them with a ``,``
 character.
 
+You can customize the output by providing an optional closure that returns the desired string. For example, to pretty
+print a map as JSON::
+
+    import static groovy.json.JsonOutput.toJson
+    import static groovy.json.JsonOutput.prettyPrint
+
+    workflow {
+        Channel.of(
+            [
+                id: 'test',
+                samples: [
+                    [ id: 'S1', path: '/path/to/s1' ],
+                    [ id: 'S2', path: '/path/to/s2' ],
+                    [ id: 'S3', path: '/path/to/s3' ]
+                ]
+            ]
+        ).dump(tag: 'map', {
+            def json = toJson(it)
+            prettyPrint( json )
+        })
+    }
```

Comment:
Even it's a nice trick, not sure we should promote this in the docs. It could make sense to add a pretty print option in the operator

Reply:
The main "problem" is, if I'm not wrong, that the argument for the closure can be any type

It can be a Map, a String, a List ...

Reply:
it should as simple as adding an option [here](https://github.com/nextflow-io/nextflow/blob/3c06a598e801611faabdbb61dfa3544b47657a88/modules/nextflow/src/main/groovy/nextflow/extension/DumpOp.groovy#L54-L54)

Reply:
Yes, I know it, but what logic needs to implement in this method due to the argument can be any object?

Maybe we can implement a default prettyPrint similar to this: https://github.com/nextflow-io/nextflow/issues/3159#issuecomment-1232864270 but not sure if with this we cover all cases or it's very specific to this

Reply:
What if we pass the target object to `JsonOutput.prettyPrint` ?